components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -1,110 +1,531 @@
+        auto tau_w_g_icol = ekat::subview(aero_tau_g_sw, icol);
+        // tau_w_f: aero_tau_forward
+        // forward scattered fraction * tau * w
+        auto tau_w_f_icol = ekat::subview(aero_tau_forward, icol);
+        // tau: aerosol
+        // aerosol extinction optical depth
+        auto tau_icol = ekat::subview(aero_tau_sw, icol);
+
+        auto work_icol = ekat::subview(work, icol);
+
+        // fetch column-specific subviews into aerosol prognostics
+        mam4::Prognostics progs =
+            mam_coupling::aerosols_for_column(dry_aero, icol);
+
+        mam4::aer_rad_props::aer_rad_props_sw(
+            team, dt, progs, atm, zi, pint, pdel, pdeldry, ssa_cmip6_sw_icol,
+            af_cmip6_sw_icol, ext_cmip6_sw_icol, tau_icol, tau_w_icol,
+            tau_w_g_icol, tau_w_f_icol, aerosol_optics_device_data, aodvis(icol), work_icol);
+
+        team.team_barrier();
+        mam4::aer_rad_props::aer_rad_props_lw(
+            team, dt, progs, atm, pint, zi, pdel, pdeldry, ext_cmip6_lw_icol,
+            aerosol_optics_device_data, odap_aer_icol);
+      });
+  Kokkos::fence();
+  // TODO: We will need to generate optical inputs files with  band ordering  that is consistent with
+  // RRTMGP
+  // Optical files depend on the band ordering in RRTMG. As a temporary fix,
+  // we are correcting the band ordering of mam4xx's ouputs, so that they are consistent with
+  // inputs in RRTMGP.
+  // Mapping from old RRTMG sw bands to new band ordering in RRTMGP
+  // than rrtmgp mam4 layout: (ncols, nswlands, nlevs +1  ) rrtmgp in emaxx:
+  // (ncols, nswlands, nlevs) Here, we copy data from kk=1 in mam4xx
+  // Here, we are following: E3SM/components/eam/src/physics/rrtmgp
+  ///cam_optics.F90
+  const auto& get_idx_rrtmgp_from_rrtmg_swbands = get_idx_rrtmgp_from_rrtmg_swbands_;
+  // postprocess output
+  Kokkos::parallel_for("postprocess", policy, postprocess_);
+  Kokkos::fence();
+
+  // nswbands loop is using rrtmg indexing.
+  Kokkos::parallel_for(
+      "copying data from mam4xx to eamxx",
+      Kokkos::MDRangePolicy<Kokkos::Rank<3> >({0, 0, 0},
+                                              {ncol_, nswbands_, nlev_}),
+      KOKKOS_LAMBDA(const int icol, const int iswband, const int kk) {
+        // Extract single scattering albedo from the product-defined fields
+        if (aero_tau_sw(icol, iswband, kk + 1) > zero) {
+          aero_ssa_sw_eamxx(icol, get_idx_rrtmgp_from_rrtmg_swbands(iswband), kk) =
+          aero_tau_ssa_sw(icol, iswband, kk + 1)/aero_tau_sw(icol, iswband, kk + 1);
+        } else {
+          aero_ssa_sw_eamxx(icol, get_idx_rrtmgp_from_rrtmg_swbands(iswband), kk) = one;
+        }
+        // Extract assymmetry parameter from the product-defined fields
+        if (aero_tau_ssa_sw(icol, iswband, kk + 1) > zero ) {
+          aero_g_sw_eamxx(icol, get_idx_rrtmgp_from_rrtmg_swbands(iswband), kk) =
+            aero_tau_g_sw(icol, iswband, kk + 1)/aero_tau_ssa_sw(icol, iswband, kk + 1) ;
```

Comment:
As above, if `aero_tau_g_sw` is actually `tau * ssa * g`, I'd prefer if we called it `aero_tau_ssa_g_sw` to make this more clear.

Reply:
Note also, this is the convention we've used in the fortran interfaces as well in the past. E.g., from RRTMG:
```
    ! Aerosol radiative properties
    real(r8) :: aer_tau    (pcols,0:pver,nbndsw) ! aerosol extinction optical depth
    real(r8) :: aer_tau_w  (pcols,0:pver,nbndsw) ! aerosol single scattering albedo * tau
    real(r8) :: aer_tau_w_g(pcols,0:pver,nbndsw) ! aerosol assymetry parameter * w * tau
    real(r8) :: aer_tau_w_f(pcols,0:pver,nbndsw) ! aerosol forward scattered fraction * w * tau
```

Reply:
> Note also, this is the convention we've used in the fortran interfaces as well in the past. E.g., from RRTMG:
> 
> ```
>     ! Aerosol radiative properties
>     real(r8) :: aer_tau    (pcols,0:pver,nbndsw) ! aerosol extinction optical depth
>     real(r8) :: aer_tau_w  (pcols,0:pver,nbndsw) ! aerosol single scattering albedo * tau
>     real(r8) :: aer_tau_w_g(pcols,0:pver,nbndsw) ! aerosol assymetry parameter * w * tau
>     real(r8) :: aer_tau_w_f(pcols,0:pver,nbndsw) ! aerosol forward scattered fraction * w * tau
> ```

I'll use these convention to name these variables.